### ISSUE TITLE
Unable to commit on the first attempt in a new repo

### ISSUE THREAD

**Issue by PavelLaptev** [gitbutlerapp/gitbutler#4346]
### Version

nightly/0.5.600

### Operating System

Mac OS X

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

When you add a repository for the very first time in GitButler and then try to commit your changes, the first attempt will fail and show you this message:

<img width="588" alt="image" src="https://github.com/user-attachments/assets/31bba594-0497-4a0e-b0d6-a433cb945fe1">

However, on the second attempt, you will be able to commit. This issue occurs only once for fresh repositories.

### How to reproduce

1. Add new repo to GitButler. Make sure your repo wasn't added to GitButler before, otherwise you wouldn't see the error.
2. Make some changes and click `Start commit`.
3. Write a commit massage and click `Commit`
4. You should see the error `Errors.commit.signing_failed`. Note: `Commit signing` in the settings was off.

Here is the video capture:

https://github.com/user-attachments/assets/424df3ef-78a1-4d10-94e0-32e0335fd263

### Expected behavior

There shouldn't be any errors on the first commit

### Relevant log output

```shell
[Error] ipc->commit_virtual_branch: {"projectId":"78d2895c-1dcd-4f91-907a-372f1d691dee","branch":"bbaced9a-3966-45a5-bbad-abd6d6507238","message":"changes","ownership":"README.md:1-2-3553f0dc951a38695e10ec6acc0083b6","runHooks":false} (2)
Error: Errors.commit.signing_failed
{code: "errors.commit.signing_failed", message: "errors.commit.signing_failed"}
	(anonymous function) (ipc.ts:35)
```
```

**Comment by krlvi** [comment]
This report is from `nightly/0.5.600`, and the commit-signing flow has changed substantially since then. In [0.14.20](https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.14.20), GitButler started reading Git's signing configuration directly and treating `gitbutler.signCommits` as an override via https://github.com/gitbutlerapp/gitbutler/pull/8395. Signing failures now also explain that signing was disabled instead of only showing the old `Errors.commit.signing_failed` message.

I'm closing this as obsolete. If a fresh repository still fails on its first commit in the current release, please open a new issue with the full error and logs from that first attempt so we can see why signing was attempted.
